pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -8,14 +8,25 @@ import (
 	"crypto/x509"
 	"encoding/pem"
 
+	"github.com/golang-jwt/jwt/v4"
+
 	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/common"
 )
 
-// AuthConfig is an asset that generates ECDSA public and private keys.
+// AuthConfig is an asset that generates ECDSA public/private keys, JWT token.
 type AuthConfig struct {
-	PublicKey, PrivateKey string
+	PublicKey, PrivateKey, Token string
 }
 
+// LocalJWTKeyType suggests the key type to be used for the token.
+type LocalJWTKeyType string
+
+const (
+	// InfraEnvKey is used to generate token using infra env id.
+	InfraEnvKey LocalJWTKeyType = "infra_env_id"
+)
+
 var _ asset.WritableAsset = (*AuthConfig)(nil)
```

Comment:
Does this actually need to be a WritableAsset? It seems like we could get rid of the `Load()`/`Files()` methods and be a regular Asset.